src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -396,20 +406,33 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
             static ImmutableArray<Diagnostic> FilterApplicableDiagnostics(
```

Comment:
💭 This could also be implemented using a local hash set to hold the equivalence keys for the current `CodeFixProvider`, and promoting the local results to `equivalenceKeys` at the end of the method.